File: compatibility-tests/src/test/java/nnl/rocks/kactoos/scalar/StickyScalarTest.java
/**
 * The MIT License (MIT)
 * <p>
 * Copyright (c) 2017-2018 Yegor Bugayenko
 * <p>
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * <p>
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 * <p>
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package nnl.rocks.kactoos.scalar;

import java.security.SecureRandom;
import nnl.rocks.kactoos.Scalar;
import org.hamcrest.MatcherAssert;
import org.hamcrest.Matchers;
import org.junit.Test;

/**
 * Test case for {@link StickyScalar}.
 *
 * @author Yegor Bugayenko (yegor256@gmail.com)
 * @version $Id: 1e6ef25b0cb50876984e8a6e85d925241d0a34b8 $
 * @since 0.4
 * @checkstyle JavadocMethodCheck (500 lines)
 */
public final class StickyScalarTest {

    @Test
    public void cachesScalarResults() throws Exception {
        final Scalar<Integer> scalar = new StickyScalar<>(
            () -> new SecureRandom().nextInt()
        );
        MatcherAssert.assertThat(
            scalar.value() + scalar.value(),
            Matchers.equalTo(scalar.value() + scalar.value())
        );
    }

}

File: kactoos-jvm/src/test/kotlin/nnl/rocks/kactoos/scalar/StickyScalarTest.kt

package nnl.rocks.kactoos.scalar

import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.Test

import java.security.SecureRandom

/**
 * Test case for [StickyScalar].
 *
 * @author Yegor Bugayenko (yegor256@gmail.com)
 * @version $Id: 1e6ef25b0cb50876984e8a6e85d925241d0a34b8 $
 * @since 0.4
 * @checkstyle JavadocMethodCheck (500 lines)
 */
class StickyScalarTest {

    @Test
    @Throws(Exception::class)
    fun cachesScalarResults() {
        val scalar = StickyScalar<Int>(
                { SecureRandom().nextInt() }
        )
        MatcherAssert.assertThat(
                scalar.value() + scalar.value(),
                Matchers.equalTo<Int>(scalar.value() + scalar.value())
        )
    }
}
